recipes/openai_terms.py

```diff
@@ -0,0 +1,122 @@
+import os
+import time
+from pathlib import Path
+from typing import Callable, List, TypeVar
+
+import httpx
+import jinja2
+import prodigy
+import prodigy.components.db
+import prodigy.components.preprocess
+import prodigy.util
+import rich
+import srsly
+from dotenv import load_dotenv
+from prodigy.util import msg
+from rich.panel import Panel
+from rich.pretty import Pretty
+
+_ItemT = TypeVar("_ItemT")
+
+DEFAULT_PROMPT_PATH = Path(__file__).parent.parent / "templates" / "terms_prompt.jinja2"
+
+# Set up openai
+load_dotenv()  # take environment variables from .env.
+
+
+def _load_template(path: Path) -> jinja2.Template:
+    # I know jinja has a lot of complex file loading stuff,
+    # but we're not using the inheritance etc that makes
+    # that stuff worthwhile.
+    if not path.suffix == ".jinja2":
+        msg.fail(
+            "The --prompt-path (-p) parameter expects a .jinja2 file.",
+            exits=1,
+        )
+    with path.open("r", encoding="utf8") as file_:
+        text = file_.read()
+    return jinja2.Template(text)
+
+
+def _parse_terms(completion):
+    # Sometimes it only returns a single item. For example, when there are
+    # many, many seeds around.
+    if "\n" not in completion:
+        return [completion.replace("-", "").strip().lower()]
+    # Other times we cannot assume the final item will have had sufficient 
+    # tokens available to complete the term, so we have to discard it.
+    lines = completion.split("\n")
```

Comment:
Do we need a `strip()` here? Or are we sure there can't be trailing newlines?

Reply:
I suppose `.lstrip()` should suffice.

Reply:
I just tried with just "lstrip" and found whitespaces at the end. Will stick to `.strip()`.